octoprint_PrintTimeGenius/analyzers/analyze_gcode_comments.py

```diff
@@ -84,7 +84,7 @@ def process_slic3r_print_time(gcode_line):
 def process_slic3r_print_time_remaining(gcode_line):
   """Match a Slic3r PE print time remaining estimate"""
   ret = dd()
-  m = re.match('\s*M73\s+P([0-9.]+)\s+R([0-9.]+)\s*', gcode_line)
+  m = re.match('\s*M73\s+(?:P([0-9.]+)\s+)?R([0-9.]+)\s*', gcode_line)
   if m:
     minutes_text = m.group(2)
```

Comment:
Does this need to be changed to `1`?

Reply:
No, I specifically used a non capturing group, which starts with `(?:` to make sure no other changes were required.

![image](https://user-images.githubusercontent.com/7568570/186121633-4a607fe2-35ec-472b-a486-53421a9c9bd6.png)

As you see in all cases the 'group 2' match is 43, which is desired.

If you want to play around with it, here's a link: https://regex101.com/r/STELrq/1

Also, the `\s*` at the end is not really required, but it does no harm, so I decided to leave it in place

Reply:
Ah, right!  I forgot about how nested parenthesis works.  Good job and very nice explanation.  I am convinced!

I think that the `\s*` is necessary at the end and also at the front because I used `re.match` and not `re.search`.  If I had used search then I agree it is unnecessary.

Thanks for your efforts!

Reply:
Read the docs a bit more on match vs search and I think you're right. I'm used to a different regex engine where you always have to use `^` to assert position at the biginning of a string, but python seems to do that by default when you use match. Good catch